content/client-lib-development-guide/features.textile

```diff
@@ -152,7 +150,7 @@ h3(#rest-channel). Channel
 ** @(RSL1c)@ When an array of @Message@ objects is provided, a single request is made to Ably. When publishing multiple messages, this approach is more efficient. However, a yet to be implemented feature should limit the total number of messages bundled in a single POST based on the default max request size, and would raise an exception if any single message exceeds that limit.
 ** @(RSL1d)@ Indicates an error if the message was not successfully published to Ably
 ** @(RSL1e)@ Allows @name@ and or @data@ to be @null@.  If any of the values are null, then key is not sent to Ably i.e. a payload with a @null@ value for @data@ would be sent as follows @{ "name": "click" }@
-** @(RSL1f)@ Unauthenticated clients without a @clientId@ (any @clientId@ is permitted):
+** @(RSL1f)@ Unauthenticated clients using basic auth without a @clientId@ (i.e. any @clientId@ is permitted):
```

Comment:
> I thought we had agreed a token without a clientId meant that a client could not use any clientId

I don't think so. As a matter of principle, there shouldn't be anything that you can do with basic auth that you can't do with token auth. We originally had tokens with a `'*'` wildcard `clientId` but we agreed to remove the wildcard, so now a token with no `clientId` string can act on behalf of any clientId, as if using basic auth.

> either way we should a test in realtime for this.

yes